debugger/lldb-extension/lldb_sgx_plugin.py

```diff
@@ -0,0 +1,313 @@
+    else:
+        print ("Can't access memory at {0:x}.".format(int(addr)) + "\n" + str(os.error))
+        return None
+
+def read_int_from_memory(addr, size):
+    mv = read_from_memory(addr, size)
+    return int.from_bytes(mv, 'little')
+
+def target_path_to_host_path(target_path):
+    so_name = os.path.basename(target_path)
+    strpath = gdb.execute("show solib-search-path", False, True)
+    path = strpath.split()[-1]
+    strlen = len(path)
+    if strlen != 1:
+        path = path[0:strlen-1]
+    host_path = path + "/" + so_name
+    return host_path
+
+def load_enclave_symbol(enclave_path, enclave_base_addr):
+    """Load enclave symbol file into current debug session"""
+
+    if os.path.exists(enclave_path) == True:
+        enclave_path = os.path.abspath(enclave_path)
+    else:
+        enclave_path = target_path_to_host_path(enclave_path)
+
+    lldb.debugger.HandleCommand("target modules add " + enclave_path)
+    lldb.debugger.HandleCommand("target modules load --file " + enclave_path
+                                + " -s " + str(enclave_base_addr))
+
+    # Store the oe_enclave address to global set that will be cleanup on exit.
+    global g_loaded_oe_enclave_addrs
+    g_loaded_oe_enclave_addrs.add(enclave_base_addr)
+    return True
+
+def unload_enclave_symbol(enclave_path, enclave_base_addr):
+    if os.path.exists(enclave_path) == True:
+        enclave_path = os.path.abspath(enclave_path)
+    else:
+        enclave_path = target_path_to_host_path(enclave_path)
+
+    target = lldb.debugger.GetSelectedTarget()
+    module = target.FindModule(lldb.SBFileSpec(enclave_path))
+    target.RemoveModule(module)
+
+    global g_loaded_oe_enclave_addrs
+    g_loaded_oe_enclave_addrs.discard(enclave_base_addr)
+
+def set_tcs_debug_flag(tcs_addr):
+    string = read_from_memory(tcs_addr + 8, 4)
+    if string is None:
+        return False
+    flag = struct.unpack('I', string)[0]
+    flag |= 1
+
+    process = lldb.debugger.GetSelectedTarget().GetProcess()
+    pid = process.GetProcessID();
+    fd = os.open("/proc/" + str(pid) + "/mem", os.O_WRONLY)
+    os.lseek(fd, int(tcs_addr + 8), 0);
+    result = os.write(fd, struct.pack('I', flag));
+    os.close(fd)
+    if result != -1:
+        return True
+    else:
+        print ("Can't access memory at {0:x}.".format(int(addr)) + "\n" + str(os.error))
+        return None
+
+def enable_oeenclave_debug(oe_enclave_addr):
+    """For a given OE enclave, load its symbol and enable debug flag for all its TCS"""
+    enclave = oe_debug_enclave_t(oe_enclave_addr)
+
+    # Check if magic matches
+    if not enclave.is_valid():
+        return False
+
+    # No version specific checks.
+    # The contract will be extended in backwards compatible manner.
+    # Debugger may use version to take specific actions in future.
+
+    # Check if debugging is enabled.
+    if enclave.debug == 0:
+        print ("oegdb: Debugging not enabled for enclave %s" % enclave.path)
```

Comment:
Should it be `oelldb` instead? Here and the below.

Reply:
I need to do a string replace.